app/src/main/java/com/teamltt/carcare/activity/CarInfoEditActivity.java

```diff
@@ -0,0 +1,90 @@
+package com.teamltt.carcare.activity;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import com.teamltt.carcare.R;
+import com.teamltt.carcare.database.DbHelper;
+import com.teamltt.carcare.model.Vehicle;
+
+public class CarInfoEditActivity extends AppCompatActivity {
+
+    private static final String TAG = "CarInfoEditActivity";
+
+    private boolean edited = false;
+    private DbHelper dbHelper;
+    long vehicleId;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_car_info_edit);
+
+        vehicleId = getIntent().getLongExtra(CarInfoActivity.EXTRA_VEHICLE_ID, -1);
+        dbHelper = new DbHelper(CarInfoEditActivity.this);
+        if (vehicleId == -1) {
+            Log.e(TAG, "invalid vehicle id");
+            finish();
+        }
+        Vehicle vehicle = dbHelper.getVehicle(vehicleId);
+        if (vehicle == null || vehicleId == -1) {
+            Log.e(TAG, "could not get vehicle properly");
+        } else {
+            updateUi(vehicle);
+        }
+    }
+
+
+    public void saveInfo(View view) {
+        Log.i(TAG, "saveInfo");
+        edited = true;
+        back(view);
+    }
+
+    public void back(View view) {
```

Comment:
Added, I think we may want to call this cancel or something since Back is a word used in the Android OS